eng/common/scripts/TypeSpec-Project-Process.ps1

```diff
@@ -0,0 +1,115 @@
+# For details see https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/TypeSpec-Project-Scripts.md
+
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $TypeSpecProjectDirectory, # A directory of `tspconfig.yaml` or a remoteUrl of `tspconfig.yaml`
+  [Parameter(Position = 1)]
+  [string] $CommitHash,
+  [Parameter(Position = 2)]
+  [string] $RepoUrl
+)
+
+. $PSScriptRoot/common.ps1
+if ($RepoUrl -and !$RepoUrl.StartsWith("https://github.com") -and ($RepoUrl.Split("/").Length -lt 4)) {
+  Write-Host "Parameter 'RepoUrl' has incorrect value: $RepoUrl. It should be similar like 'https://github.com/Azure/azure-rest-api-specs'"
+  exit
+}
+
+function CreateUpdate-TspLocation() {
+  $serviceDir = ""
+  $additionalDirs = @()
+
+  # Parse tspcofig.yaml to get service-dir, additionalDirectories, and package-dir
+  if ($tspConfigYaml["parameters"] -and $tspConfigYaml["parameters"]["service-dir"]) {
+    $serviceDir = $tspConfigYaml["parameters"]["service-dir"]["default"];
+  }
+  else {
+    Write-Host "Missing service-dir in parameters section of tspconfig.yaml."
+    exit
+  }
+  if ($tspConfigYaml["dependencies"] -and $tspConfigYaml["dependencies"]["additionalDirectories"]) {
+    $additionalDirs = $tspConfigYaml["dependencies"]["additionalDirectories"];
+  }
+
+  $packageDir = Get-PackageDir
+
+  $repoRootPath =  (Join-Path $PSScriptRoot .. ..)
+  $repoRootPath = Resolve-Path $repoRootPath
+  $repoRootPath = $repoRootPath -replace "\\", "/"
+
+  # Create service-dir if not exist
+  $serviceDir = Join-Path $repoRootPath "sdk" $serviceDir
+  if (!(Test-Path -Path $serviceDir)) {
+    New-Item -Path $serviceDir -ItemType Directory
+    Write-Host "created service folder $serviceDir"
+  }
+
+  # Create package-dir if not exist
+  $packageDir = Join-Path $serviceDir $packageDir
+  if (!(Test-Path -Path $packageDir)) {
+    New-Item -Path $packageDir -ItemType Directory
+    Write-Host "created package folder $packageDir"
+  }
+
+  # Load tsp-location.yaml if exist
+  $tspLocationYamlPath = Join-Path $packageDir "tsp-location.yaml"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    $tspLocationYaml = Get-Content -Path $tspLocationYaml -Raw | ConvertFrom-Yaml
+  }
+
+  # Update tsp-location.yaml
+  $tspLocationYaml["commit"] = $CommitHash
+  $repo = ($RepoUrl -split "/")[3..4] -join "/"
+  $tspLocationYaml["repo"] = $repo
+  $tspLocationYaml["directory"] = $TypeSpecProjectDirectory
+  $tspLocationYaml["additionalDirectories"] = $additionalDirs
```

Comment:
What type is additionaDirs? Is it an array or a string? Will this work if there is more than one?